tools/runners/parallels.sh

```diff
@@ -0,0 +1,410 @@
+  local aborted_tests=$(cat $curr_report_file | grep "ABORTED:" | awk '{print $2}')
+  local skipped_tests=$(cat $curr_report_file | grep "SKIPPED:" | awk '{print $2}')
+
+  ((_overall_test_number+=total_tests))
+  ((_total_passed+=passed_tests))
+  ((_total_failed+=failed_tests))
+  ((_total_aborted+=aborted_tests))
+  ((_total_skipped+=skipped_tests))
+
+  if [[ $passed_tests == 1 ]]; then
+    echo -e "$dir_number:\t$test_target\tPASSED" >> $overall_report_file
+  elif [[ $failed_tests == 1 ]]; then
+    echo -e "$dir_number:\t$test_target\tFAILED" >> $overall_report_file
+  elif [[ $aborted_tests == 1 ]]; then
+    echo -e "$dir_number:\t$test_target\tABORTED" >> $overall_report_file
+  elif [[ $skipped_tests == 1 ]]; then
+    echo -e "$dir_number:\t$test_target\tSKIPPED" >> $overall_report_file
+  else
+    echo -e "$dir_number:\t$test_target\tPARSING ERROR" >> $overall_report_file
+    return 1
+  fi
+
+  return 0
+}
+
+function generate_total_report {
+  env_dir=$1; shift
+
+  testing_report_dir=$env_dir/TestingReports
+  mkdir $testing_report_dir
+
+  total_number_of_tests=$(find $env_dir/*/TestingReports/* -maxdepth 0 -type d | wc -l)
+
+  overall_report_file=$testing_report_dir/Report.txt
+
+  touch $overall_report_file
+
+  for worker in $(ls $env_dir | grep _worker)
+  do
+    if [ ! -d $env_dir/$worker/TestingReports ]; then
+      continue;
+    fi
+    for item in $(ls $env_dir/$worker/TestingReports)
+    do
+      local abs_path=$env_dir/$worker/TestingReports/$item
+      local dir_number=$(basename $(find $abs_path -maxdepth 1 -type d | sort -r | head -n 1))
+      local dir_number_lead_zeroes=$(printf "%0${#total_number_of_tests}d" $dir_number)
+      process_report $abs_path/Report.txt $dir_number_lead_zeroes
+      process_report_status=$?
+
+      local current_test_dirname=$testing_report_dir/$dir_number_lead_zeroes
+      mv $abs_path/$dir_number $current_test_dirname
+      if [ $process_report_status != 0 ]; then
+        mv $abs_path/Report.txt $current_test_dirname/
+      fi
+      rm -r $abs_path
+    done
+    rm -r $env_dir/$worker/TestingReports
+  done
+
+  sort -o $overall_report_file $overall_report_file
+
+  echo "$(echo ${LINE2} | cat - $overall_report_file)" > $overall_report_file
+  echo "$(echo 'Test target:' $_testfile | cat - $overall_report_file)" > $overall_report_file
+  echo "$(echo ${LINE1} | cat - $overall_report_file)" > $overall_report_file
+
+  echo ${LINE2} >> $overall_report_file
+  echo "TOTAL: $_overall_test_number" >> $overall_report_file
+  echo "PASSED: $_total_passed" >> $overall_report_file
+  echo "FAILED: $_total_failed" >> $overall_report_file
+  echo "ABORTED: $_total_aborted" >> $overall_report_file
+  echo "SKIPPED: $_total_skipped" >> $overall_report_file
+  echo ${LINE2} >> $overall_report_file
+  echo "Execution time:" $(seconds2time $(($ts_finish - $ts_start))) >> $overall_report_file
+  echo ${LINE1} >> $overall_report_file
```

Comment:
```suggestion
  echo "$(echo ${LINE2} | cat - $overall_report_file)" > $overall_report_file
  echo "$(echo 'Test target:' $_testfile | cat - $overall_report_file)" > $overall_report_file
  echo "$(echo ${LINE1} | cat - $overall_report_file)" > $overall_report_file

  echo ${LINE2} >> $overall_report_file
  echo "TOTAL: $_overall_test_number" >> $overall_report_file
  echo "PASSED: $_total_passed" >> $overall_report_file
  echo "FAILED: $_total_failed" >> $overall_report_file
  echo "ABORTED: $_total_aborted" >> $overall_report_file
  echo "SKIPPED: $_total_skipped" >> $overall_report_file
  echo ${LINE2} >> $overall_report_file
  echo "Execution time:" $(seconds2time $(($ts_finish - $ts_start))) >> $overall_report_file
  echo ${LINE1} >> $overall_report_file
```
I could see it being useful to print this information to the console as well.

Reply:
@jacobkeeler Done in 0c96eea